controlplane/src/core/repositories/FederatedGraphRepository.ts

```diff
@@ -782,4 +788,188 @@ export class FederatedGraphRepository {
         }) as GraphApiKeyDTO,
     );
   }
+
+  public async registerClient(userId: string, federatedGraphName: string, clientName: string): Promise<string> {
+    if (!clientName) {
+      throw new Error('client name is empty');
+    }
+    const graph = await this.db.query.targets.findFirst({
```

Comment:
This should be done before. Once in the RPC handler.

Reply:
Done